src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -471,15 +471,20 @@ bool Terminal::SendKeyEvent(const WORD vkey,
 
     _StoreKeyEvent(vkey, scanCode);
 
-    // As a Terminal we're mostly interested in getting key events from physical hardware (mouse & keyboard).
-    // We're thus ignoring events whose values are outside the valid range and unlikely to be generated by the current keyboard.
-    // It's very likely that a proper followup character event will be sent to us.
-    // This prominently happens using AutoHotKey's keyboard remapping feature,
-    // which sends input events whose vkey is 0xff and scanCode is 0.
-    // We need to check for this early, as _CharacterFromKeyEvent() always returns 0 for such invalid values,
-    // making us believe that this is an actual non-character input, while it usually isn't.
+    // Certain applications like AutoHotKey and its keyboard remapping feature,
+    // send us key events using SendInput() whose values are outside of the valid range.
     // GH#7064
-    if (vkey == 0 || vkey >= 0xff || scanCode == 0)
+    if (vkey == 0 || vkey >= 0xff)
+    {
+        return false;
+    }
+
+    // While not explicitly permitted, a wide range of software, including Windows' own touch keyboard,
+    // sets the wScan member of the KEYBDINPUT structure to 0, resulting in scanCode being 0 as well.
+    // --> Alternatively get the scanCode from the vkey if possible.
+    // GH#7495
+    const auto sc = scanCode ? scanCode : _ScanCodeFromVirtualKey(vkey);
+    if (sc == 0)

```

Comment:
I've been thinking for a while whether I should allow `sc` to be 0 or not. I've decided for the latter, because:
WT is already broken and if we un-break WT by just enough for this to be fixed, we can keep more defensive mechanisms, allowing us to rely more on certain properties, like... `scanCode` never being zero in `TerminalInput`.

I think it should be fine to keep this if-condition. 🙂 (...until some weird Lumia 950 XL with full fledged Windows 10 Pro proves us wrong 😂)

Reply:
I think it's important I mention the reason we need `scanCode`s here again (because this topic is kinda complicated)...
When we receive a key event we must decide whether we consume the event or not. If we do consume it we'll not receive a follow-up character event no matter what. We'd like to only consume non-character key events, because the character events contain proper unicode characters and the key events don't.
In order to decide this we use `ToUnicodeEx` which translates a `vkey` and some modifier keys to one or more characters. If it spits out characters we ignore the key and wait for an appropriate character event.
Unfortunately there can be multiple meanings to a single `vkey` depending on the keyboard state and depending on the key in question that was pressed. Two or more keys might map to the same `vkey` but have different `scanCode`s.
Due to this `ToUnicodeEx` asks for a `scanCode` allowing it to correctly map unicode characters.

Reply:
This is an excellent description. Thanks!